genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobKickoffTask.java

```diff
@@ -203,15 +203,17 @@ protected synchronized void createUser(final String user, final String group) th
         } catch (final IOException ioe) {
             log.debug("User does not exist. Creating it now.");
 
-            // Create the group for the user.
-            if (group != null) {
+            // Create the group for the user if its not the same as the user.
+            if (group != null && !group.equals(user) && StringUtils.isNotBlank(group)) {
```

Comment:
The check for null and the call to StringUtils.isNotBlank are redundant. IsNotBlank checks for null so I would just put that first.